### PR TITLE
Uncertainty: Conformal Prediction V1.2 - move conformal evaluation performance metrics inside of Conformal class in conformal.py

### PR DESCRIPTION
**Pull request by Kevin-Chen0**
## :microscope: Background

- The current conformal evaluation performance metrics `interval_width` and `miscoverage_rate` are in conformal prediction tutorial [notebook](https://github.com/ourownstory/neural_prophet/blob/main/tutorials/feature-use/uncertainty_conformal_prediction.ipynb), making it too bloated. This is part of #1101.

## :crystal_ball: Key changes

- Move `interval_width` and `miscoverage_rate` conformal evaluation performance metrics inside of `Conformal` class in the conformal.py.

## :clipboard: Review Checklist
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, added docstrings and data types to function definitions.
- [x] I have added pytests to check whether my feature / fix works.

Please make sure to follow our best practices in the [Contributing guidelines](https://github.com/ourownstory/neural_prophet/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: neuralprophet/conformal.py [hunk 1]
```diff
@@ -190,3 +190,62 @@ def plot(self, plotting_backend: str):
...
+
+        Parameters
+        ----------
+            df : pd.DataFrame
+                test dataframe with the conformal prediction intervals
+
+        Returns
+        -------
+            pd.DataFrame
+                table containing evaluation metrics such as interval_width and miscoverage_rate
```
Comment (leoniewgnr) [inline neuralprophet/conformal.py:205]: Does is not only return interval_width and miscoverage_rate? Maybe just specify: 
```suggestion
                table containing the two evaluation metrics interval_width and miscoverage_rate for each predicted interval
```

File: neuralprophet/conformal.py [hunk 2]
```diff
@@ -190,3 +190,62 @@ def plot(self, plotting_backend: str):
...
+                        <= row["y"]
+                        <= row[f"yhat{step_number} + qhat{step_number}"]
+                    ),
+                    axis=1,
+                )
+                coverage_rate = n_covered.sum() / len(df)
+                miscoverage_rate = 1 - coverage_rate
+            elif self.method == "cqr":
+                quantile_hi = str(max(quantiles) * 100)
+                quantile_lo = str(min(quantiles) * 100)
```
Comment (leoniewgnr) [inline neuralprophet/conformal.py:226]: Did you mean self.quantiles here instead of quantiles?

**Review by leoniewgnr — CHANGES_REQUESTED** [review]
Overall, it looks good to me! There need to be done some changes because some variables are not defined and some tests failed.

File: neuralprophet/conformal.py (hunk 2 above)
Comment (Kevin-Chen0) [reply neuralprophet/conformal.py]: I fixed, great catch!

File: neuralprophet/conformal.py [hunk 3]
```diff
@@ -190,3 +190,62 @@ def plot(self, plotting_backend: str):
...
+                        row[f"yhat{step_number} - qhat{step_number}"]
+                        <= row["y"]
+                        <= row[f"yhat{step_number} + qhat{step_number}"]
+                    ),
+                    axis=1,
+                )
+                coverage_rate = n_covered.sum() / len(df)
+                miscoverage_rate = 1 - coverage_rate
+            elif self.method == "cqr":
+                quantile_hi = str(max(quantiles) * 100)
```
[the remark this answers is not in the document]
Comment (Kevin-Chen0) [reply neuralprophet/conformal.py]: I fixed, great catch!

**Review by leoniewgnr — APPROVED** [review]
Looks good to me!

**Pull request merged**